types/pluck.d.ts

```diff
@@ -1,10 +1,10 @@
 import { Placeholder } from 'ramda';
 
 export function pluck<K extends PropertyKey>(prop: K extends Placeholder ? never : K): {
-  <U extends readonly unknown[] | Record<K, any>>(obj: Record<PropertyKey, U>): U extends readonly (infer T)[] ? T[] : U extends Record<K, infer T> ? T[] : never;
+  <U extends Record<K, any>, IK extends string>(obj: Record<IK, U>): { [KK in keyof typeof obj]: U[K] };
```

Comment:
I think this will be a bit easier to understand like this:
```suggestion
  <U extends Record<K, any>, O extends Record<PropertyKey, U>>(obj: O): Record<keyof O, U[K]>;
```

All 3 locations can be updated as such

Reply:
Yes, your suggestion is much better. However, I took the liberty to make them a little bit more complex to cover more cases. Please, tell me what you think